### PR TITLE
Introduction of _beat directory

### PR DESCRIPTION
**Pull request by ruflin**
The _beat directory is used for various files which are for locality reason for example in a module and need to be collected to generate the end output. Currently the following documents are in the .beat directory:

* Local fields.yml
* Local config.yml
* Local docs.asciidoc
* Local Dashboards

## _beat directory
The _beat directory makes it possible to better automate and generalise the collection of this files. In addition the local directories are not cluttered with all the fields mentioned above. A dot directory was chosen so the name does not conflict with any other potential directories. As the files are local, it also allows to enhance and extend the files, for example the documentation file with links, config and data example. Like this the docs stay in sync with the development.

## Document naming

The naming of each document is generic (docs.asciidoc, fields.yml). In case a metricset is renamed, no renaming of the meta files has to happen.

The Data in data.json is collected from the system tests. This could potentially be automated in the future or have a command for it to generate the docs? Problem is that timestamps change every time.

## Additional changes

* Update Makefile and collector scripts to clean collection directory before collection to make sure old documents are removed
* Move all files which are collected by scripts to a _beat directory
* Introduction of local configuration files: config.yml
* Added documentation files for all modules and metricsets
* Added data.json files to all metricsets from system tests
* Auto generate links to modules and metricsets

In the future all fields which are used for collection should be moved under a _beat directory, for example also beat.yml files. That would mean every beat can have a .beat directory with this kind of files inside.

### CODE REVIEW

**Comment by tsg** [comment]
I find `data.json` a bit different from the others, because that is generated (in the future) as opposed to being the source for generation. But I understand the benefits of having it in the same folder so that it can be included in docs and cannot think of a better place for it, so I'm good with it.

File: metricbeat/scripts/docs_collector.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+The Apache module supports the standard configuration options which can be found
+here (add link). Below is an example of a configuration option:
+
+[source,yaml]
+----
+"""
+
+            # Load metricset yaml
+            with file(config_file) as f:
+                # Add 4 spaces for indentation in fron of each line
```
Comment (tsg) [inline metricbeat/scripts/docs_collector.py]: s/fron/front

**Comment by ruflin** [comment]
@tsg Agree, hopefully as soon as we have `data.json` generated automatically, I hope we don't even have to check it in anymore. To not have to include it directly but check it in the data directory in the future, perhaps we could use Include Files (http://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#include-files). I don't know if that works for any kind of content. So it would be generated by the docs generator.

File: metricbeat/scripts/docs_collector.py (hunk 1 above)
Comment (tsg) [inline metricbeat/scripts/docs_collector.py]: And the code seems to add only two spaces?

File: metricbeat/scripts/docs_collector.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                metricset_file += "[source,json]\n"
+                metricset_file += "----\n"
+
+                with file(data_file) as f:
+                    metricset_file += f.read()
+
+                metricset_file += "----\n"
+
+            # Write metricset docs
+            f = open(os.path.abspath("docs") + "/modules/" + module + "/" + metricset + ".asciidoc", 'w')
```
Comment (tsg) [inline metricbeat/scripts/docs_collector.py]: `with open() as f` would be better here to make sure the file is closed immediately after.

File: metricbeat/scripts/docs_collector.py (hunk 1 above)
Comment (ruflin) [inline metricbeat/scripts/docs_collector.py]: Yep, I switched to `metricbeat.modules` config option ... Fixed

Comment (ruflin) [inline metricbeat/scripts/docs_collector.py]: Fixed also for config collector

**Comment by tsg** [comment]
LGTM, left minor comments. Nice work!

File: metricbeat/scripts/docs_collector.py (hunk 2 above)
Comment (ruflin) [inline metricbeat/scripts/docs_collector.py]: Fixed

**Comment by ruflin** [comment]
@tsg Changes pushed. I added one more change so that now all generated docs have a not on top that they were generated.

**Pull request merged**